props/prop_base.h

```diff
@@ -435,12 +416,68 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #else
     SaberBase::SetVariation(0);
 #endif
+  }
 
+  // Select preset (font/style)
+  virtual void SetPreset(int preset_num, bool announce) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles(preset_num);
+    if (announce) {
+      AnnouncePreset();
+    }
+    SetBladeLength();
+    SetBladeStyle();
     if (on) On();
     if (announce) SaberBase::DoNewFont();
     TRACE(PROP, "end");
   }
 
+  // Update Blade Style (no On/Off for use in Edit Mode)
+  void UpdateStyle(int preset_num) {
```

Comment:
I don't think UpdateStyle should take a number, it should just work on the current preset.

Reply:
changed